### PR TITLE
TMDM-12514 View search show errors when views contains inherit type

### PR DESCRIPTION
**Pull request by zfkang-talend**
Jira: https://jira.talendforge.org/browse/TMDM-12514

**What is the current behavior?** (You should also link to an open issue here)
1. If field is inherit and 0 - n+, result is List<DataRecord>, have excepiton of ClassCastException when get the record.

**What is the new behavior?**
1. If field is inherit and 0 - n+, result is List<DataRecord>, if contains many result, only get the first one.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/FullTextQueryHandler.java [hunk 1]
```diff
@@ -388,7 +391,15 @@ public DataRecord next() {
...
                             } else if(typedExpression instanceof Type) {
                                 FieldMetadata fieldMetadata = ((Type) typedExpression).getField().getFieldMetadata();
-                                nextRecord.set(newField, ((DataRecord) next.get(fieldMetadata.getName())).getType().getName());
+                                Object dataRecord = next.get(fieldMetadata.getName());
+                                if (dataRecord != null) {
+                                    if (dataRecord instanceof List) {
+                                        dataRecord = (DataRecord) ((List) dataRecord).get(0);
+                                    }
+                                    nextRecord.set(newField, nextRecord.getType().getName());
+                                } else {
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/FullTextQueryHandler.java]: In Line 399, Would it be: nextRecord.set(newField, dataRecord.getName()); ?

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/FullTextQueryHandler.java [hunk 2]
```diff
@@ -388,7 +391,15 @@ public DataRecord next() {
...
+                                Object dataRecord = next.get(fieldMetadata.getName());
+                                if (dataRecord != null) {
+                                    if (dataRecord instanceof List) {
+                                        dataRecord = (DataRecord) ((List) dataRecord).get(0);
+                                    }
+                                    nextRecord.set(newField, ((DataRecord) dataRecord).getType().getName());
+                                } else {
+                                    nextRecord.set(newField, StringUtils.EMPTY);
+                                }
                             } else if (typedExpression instanceof MetadataField) {
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/hibernate/FullTextQueryHandler.java]: Looks like we have duplicate code lines: 394 to 402, and 319 to 327, is it possible to have a common private method for both?

File: org.talend.mdm.core.storage.sql/src/test/java/com/amalto/core/query/StorageFullTextTest.java [hunk 3]
```diff
@@ -1890,6 +1904,23 @@ public void testFKContainsAndContainsSentenceSearch() {
...
+        List<TypedExpression> fields = UserQueryHelper.getFields(contract, "detail/@xsi:type");
+        UserQueryBuilder qb = from(contract).select(prepareContractFields(contract)).where(fullText("1"));
+        for (TypedExpression field : fields) {
+            qb.select(field);
+        }
+        StorageResults results = storage.fetch(qb.getSelect());
+        try {
+            assertTrue(results.getSize() == 1);
+            for (DataRecord result : results) {
+                assertTrue("1".equals(result.get("id")));
```
Comment (lpw82gf) [inline org.talend.mdm.core.storage.sql/src/test/java/com/amalto/core/query/StorageFullTextTest.java]: `assertEquals("1", result.get("id"))` would be better.

File: org.talend.mdm.core.storage.sql/src/test/resources/com/amalto/core/query/metadata.xsd [hunk 4]
```diff
@@ -2793,4 +2793,49 @@
             </xsd:element>
         </xsd:all>
     </xsd:complexType>
+    <xsd:element name="Contract">
+    <xsd:complexType>
```
Comment (lpw82gf) [inline org.talend.mdm.core.storage.sql/src/test/resources/com/amalto/core/query/metadata.xsd]: indentation

**Pull request merged**